### PR TITLE
ci: improve feedback on Github pull requests.

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

File: bluebrain/repo-bluebrain/packages/py-morphio/package.py [hunk 1]
```diff
@@ -26,4 +26,4 @@ class PyMorphio(PythonPackage):
     depends_on('ninja', type='build')
     depends_on('cmake@3.2:', type='build')
     depends_on('py-numpy@1.14.1:', type='run')
-    depends_on('hdf5', type=('build', 'run'))
+    depends_on('hdf5')
```
Comment (olupton) [inline bluebrain/repo-bluebrain/packages/py-morphio/package.py:29]: Is this diff intentional?

File: share/spack/templates/reports/junit.xml [hunk 2]
```diff
@@ -22,15 +22,9 @@
         <testcase classname="{{ package.name }}"
                   name="{{ package.id }}"
                   time="{{ package.elapsed_time }}">
-<!--
-    BBP: GitLab expects any and all output in the failure tag :-/
```
Comment (olupton) [inline share/spack/templates/reports/junit.xml:26]: Did something change?

**Review by olupton — COMMENTED** [review]
Seems like quite a lot of plumbing, but it seems reasonable to test it out in the wild now.

File: share/spack/templates/reports/junit.xml (hunk 2 above)
Comment (matz-e) [reply share/spack/templates/reports/junit.xml:26]: With this, we don't have to rely on GitLab's crappy test result details any longer. It'll upload all the content there to snippets.

I'd like to keep the diff to upstream as small as possible, thus reverting the changes for GitLab here.

File: bluebrain/repo-bluebrain/packages/py-morphio/package.py (hunk 1 above)
Comment (matz-e) [reply bluebrain/repo-bluebrain/packages/py-morphio/package.py:29]: Yes, pretty much. HDF5 is not a `run` dependency, but a link one.

**Review by olupton — APPROVED** [review]

**Pull request merged**